### PR TITLE
Add twilio support

### PR DESCRIPTION
**Pull request by rachaelshaw**
This adds a new helper for sending a Twilio message. Tested in a new Sails app!

### CODE REVIEW

File: lib/private/twilio/send-message.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
+module.exports = {
+
+
+  friendlyName: 'Send message',
+
+
+  description: 'Send a message using the Twilio SMS API',
```
Comment (mikermcneil) [inline lib/private/twilio/send-message.js:7]: .

File: lib/private/twilio/send-message.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      description: 'The "auth token" associated with your Twilio account.',
+      whereToGet: {
+        url: 'https://www.twilio.com/console',
+      },
+      required: true
+    },
+
+    from: {
+      example: '+16155551234',
+      description: 'The \'From\' phone number you\'d like to use to send the SMS.',
```
Comment (mikermcneil) [inline lib/private/twilio/send-message.js:35]: `'`

File: lib/private/twilio/send-message.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+      extendedDescription: 'This phone number is assigned to you by Twilio. If omitted, the first available number will be used.',
```
Comment (mikermcneil) [inline lib/private/twilio/send-message.js:36]: required

File: lib/private/twilio/send-message.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+      required: true
+    },
+
+    to: {
+      example: '+16155556789',
+      description: 'The \'To\' phone number the message is sending to.',
```
Comment (mikermcneil) [inline lib/private/twilio/send-message.js:42]: `'`

File: lib/private/twilio/send-message.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    message: {
+      example: 'Hey there friend!',
+      description: 'The body of the message being sent',
+      extendedDescription: 'Optional if a `mediaUrl` is specified.'
+    },
+
+    mediaUrl: {
+      example: 'https://sailsjs.com/logo.png',
+      description: 'The URL of a gif, png, or jpeg image.',
+      extendedDescription: 'An alternative to `message`.'
```
Comment (mikermcneil) [inline lib/private/twilio/send-message.js:55]: wait, can I use both or not?

File: lib/private/twilio/send-message.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+  },
+
+
+  exits: {
+
+    success: {
+      outputType: 'string',
+      outputDescription: 'Twilio message ID',
+      extendedDescription: 'Note that sending the message does not necessarily mean the message was _delivered_ successfully.'
```
Comment (mikermcneil) [inline lib/private/twilio/send-message.js:66]: So sending the message means that it was received by twillio, but not necessarily that it was delivered to the recipient's phone?

File: lib/private/twilio/send-message.js [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+    },
+
+  },
+
+
+  exits: {
+
+    success: {
+      outputType: 'string',
+      outputDescription: 'Twilio message ID',
```
Comment (mikermcneil) [inline lib/private/twilio/send-message.js:65]: can we use twillio's terminology for this, to the extent it is not confusing?  (e.g. `Twillio message ID ("sid")`)

File: lib/private/twilio/send-message.js [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+      example: 'Dafd4example5852c3c69bZe5e1e704a9102',
+      description: 'The "auth token" associated with your Twilio account.',
+      whereToGet: {
+        url: 'https://www.twilio.com/console',
+      },
+      required: true
+    },
+
+    from: {
+      example: '+16155551234',
```
Comment (mikermcneil) [inline lib/private/twilio/send-message.js:34]: How does it behave if the plus sign is omitted?

Comment (mikermcneil) [reply lib/private/twilio/send-message.js:34]: Maybe a moreInfoUrl with guidance on formatting international vs. US numbers

File: lib/private/twilio/send-message.js [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+    },
+
+    to: {
+      example: '+16155556789',
+      description: 'The \'To\' phone number the message is sending to.',
+      required: true
+    },
+
+    message: {
+      example: 'Hey there friend!',
```
Comment (mikermcneil) [inline lib/private/twilio/send-message.js:47]: Would you include `type` for all of these, for consistency with the other helpers in sails-hook-organics?

File: lib/private/twilio/send-message.js (hunk 5 above)
Comment (mikermcneil) [reply lib/private/twilio/send-message.js:55]: Also, would you mind adding some more info on what this means and how it's delivered?  e.g. does Twillio say it is an MMS?  (maybe a moreInfoUrl pointing at Twillio's explanation)

File: lib/private/twilio/send-message.js [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+
+    from: {
+      example: '+16155551234',
+      description: 'The \'From\' phone number you\'d like to use to send the SMS.',
+      extendedDescription: 'This phone number is assigned to you by Twilio. If omitted, the first available number will be used.',
+      required: true
+    },
+
+    to: {
+      example: '+16155556789',
```
Comment (mikermcneil) [inline lib/private/twilio/send-message.js:41]: same thing about formatting the phone number here (moreInfoUrl would also work)

File: lib/private/twilio/send-message.js [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+    authToken: {
+      example: 'Dafd4example5852c3c69bZe5e1e704a9102',
+      description: 'The "auth token" associated with your Twilio account.',
+      whereToGet: {
+        url: 'https://www.twilio.com/console',
+      },
+      required: true
+    },
+
+    from: {
```
Comment (mikermcneil) [inline lib/private/twilio/send-message.js:33]: Does `from` and `to` match how it works in sendgrid?

File: lib/private/twilio/send-message.js [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+    accountSid: {
+      example: 'DZe5eafd3c69b1e74example5852c04a9102',
+      description: 'The "account sid" associated with your Twilio account.',
+      whereToGet: {
+        url: 'https://www.twilio.com/console',
+      },
+      required: true
+    },
+
+    authToken: {
```
Comment (mikermcneil) [inline lib/private/twilio/send-message.js:24]: can we change the input name to `secret` for consistency?

https://github.com/sailshq/sails-hook-organics/blob/e4223267dadf90c9f013c61b8de40481c91d25a7/lib/private/sendgrid/send-html-email.js#L48-L60

File: lib/private/twilio/send-message.js [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+    if (!message && !mediaUrl) {
+      throw new Error('Either a `message` or `mediaUrl` (or both) must be specified.');
+    }
+
+    // Encode our credentials.
+    var credentials = Buffer.from(`${accountSid}:${authToken}`).toString('base64');
+
+    var response = await Http.sendHttpRequest.with({
+      method: 'POST',
+      url: `Accounts/${accountSid}/Messages.json`,
```
Comment (mikermcneil) [inline lib/private/twilio/send-message.js:86]: this needs a leading slash, doesn't it?

File: lib/private/twilio/send-message.js [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+      baseUrl: `https://api.twilio.com/2010-04-01`,
+      body: {
+        'From': from,
```
Comment (mikermcneil) [inline lib/private/twilio/send-message.js:89]: Would you remove the `'` please?

File: lib/private/twilio/send-message.js (hunk 13 above)
Comment (mikermcneil) [inline lib/private/twilio/send-message.js:86]: don't forget to `encodeURIComponent()` on the accountSid

File: lib/private/twilio/send-message.js (hunk 1 above)
Comment (rachaelshaw) [reply lib/private/twilio/send-message.js]: done!

File: lib/private/twilio/send-message.js (hunk 2 above)
Comment (rachaelshaw) [reply lib/private/twilio/send-message.js]: Done

File: lib/private/twilio/send-message.js (hunk 4 above)
Comment (rachaelshaw) [reply lib/private/twilio/send-message.js]: Done

File: lib/private/twilio/send-message.js (hunk 3 above)
Comment (rachaelshaw) [reply lib/private/twilio/send-message.js]: Fixed

File: lib/private/twilio/send-message.js (hunk 5 above)
Comment (rachaelshaw) [reply lib/private/twilio/send-message.js]: Took a stab at this, would love your thoughts when you re-review. (Won't mark as resolved yet.)

File: lib/private/twilio/send-message.js (hunk 10 above)
Comment (rachaelshaw) [reply lib/private/twilio/send-message.js]: Done

File: lib/private/twilio/send-message.js (hunk 11 above)
Comment (rachaelshaw) [reply lib/private/twilio/send-message.js]: Input names match, but in sendgrid "to" is listed first; we talked about listing the things that can be globally configured first for this one, though, so I'm leaving this in the existing order at the moment. Let me know your thoughts when you review, I'll leave this one unresolved for now.

File: lib/private/twilio/send-message.js (hunk 12 above)
Comment (rachaelshaw) [reply lib/private/twilio/send-message.js]: done

File: lib/private/twilio/send-message.js (hunk 14 above)
Comment (rachaelshaw) [reply lib/private/twilio/send-message.js]: done

File: lib/private/twilio/send-message.js (hunk 13 above)
Comment (rachaelshaw) [reply lib/private/twilio/send-message.js]: yep! good call. fixed.

Comment (rachaelshaw) [reply lib/private/twilio/send-message.js]: done

**Pull request merged**